plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -86,4 +98,22 @@ public void dispose() {
     content.dispose();
     super.dispose();
   }
+
+  private DeployPreferencesPanel getPreferencesPanel(Composite container) {
+    IProject project = AdapterUtil.adapt(getElement(), IProject.class);
+    IFacetedProject facetedProject = null;
+    try {
+      facetedProject = ProjectFacetsManager.create(project);
```

Comment:
the variable declaraion and most of the code should be inside the try block so it's cleaner